In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
import cv2
import numpy as np

gt = cv2.imread('/content/drive/MyDrive/Impulse_19/Task#7/Barbara_Ground_truth.png')[:,:,0]
superResImg = cv2.imread('/content/drive/MyDrive/Impulse_19/Task#6/superResolutedImg.jpg')[:,:,0]

intermediate_img = np.zeros((256, 256))
intermediate_img[0:-1, 0:-1] = superResImg
superResImg = intermediate_img.copy()

**Function to calculate the Signal-to-Noise ratio of the Super Resoluted image obtained in the previous task in logarithmic scale (dB)**

In [ ]:
def snr(gt, superResImg):
  max_pixel = np.max(superResImg)
  mse = np.mean((gt - superResImg)**2)

  return 20 * np.log10(max_pixel/np.sqrt(mse))

In [ ]:
img_snr = snr(gt, superResImg)
print('The SNR of the super resoluted image is {} dB'.format(img_snr))

The SNR of the super resoluted image is 6.518001384313391 dB


Thus, we obtained a SNR of 6.5 dB which is on the higher side with much room for future work and improvements on the same